uwblueprint/social-diversity-for-children
# feat(class): Class Details Register Modal Flow

### Notion ticket link



[Class Details Register Modal Flow](https://www.notion.so/uwblueprintexecs/Class-Details-Register-Modal-Flow-6d7153d995c447898064a3ce05316c41)



https://user-images.githubusercontent.com/44826218/134452359-aaf75acd-0185-4304-9ce3-22f2d6b5b3df.mp4


https://user-images.githubusercontent.com/44826218/134452381-3926ea9b-b94d-4a02-a7a7-161e142c81f2.mp4


https://user-images.githubusercontent.com/44826218/134452392-9ff8bcb8-41a8-4b32-badc-8fa57cf40a9b.mp4


### Implementation description

- Used Chakra UI modal (similar to the language modal) to implement class details registration workflow
- The routing of the button depends on whether the current user has a session (routes to `/parent-enrollment` or `/login`)
- Add modal for ineligible class prompts
- Made and used several util methods for date and time range formatting to match look of Figma



### Steps to test

1. Frontend implementions satisfies designer's intentions (see video attachments)

1. Create several classes and programs with varying time range, dates
2. Click try to register by clicking on classes
3. Ensure that dates, times, class details shows up as intended and matches Figma designs



### What should reviewers focus on?

- How's the look and feel of the modals?
- Are the util methods general and robust enough?
- Are the modal used properly in the code implemention?

### Checklist
